File: src/main/java/org/irislang/jiris/core/exceptions/fatal/IrisWrongSelfException.java
package org.irislang.jiris.core.exceptions.fatal;

/**
 * Created by yuwen on 2017/7/20 0020.
 */
public class IrisWrongSelfException extends IrisFatalException {
    public IrisWrongSelfException(String fileName, int lineNumber, String message) {
        super(fileName, lineNumber, message);
    }

    @Override
    public String GetFatalExceptionName() {
        return "WrongSelfException";
    }
}

File: src/main/java/org/irislang/jiris/core/exceptions/fatal/IrisWrongSelfException.kt
package org.irislang.jiris.core.exceptions.fatal

/**
 * Created by yuwen on 2017/7/20 0020.
 */
class IrisWrongSelfException(fileName: String, lineNumber: Int, message: String) : IrisFatalException(fileName, lineNumber, message) {

    override fun GetFatalExceptionName(): String {
        return "WrongSelfException"
    }
}
